web/client/components/catalog/SharingLink.jsx

```diff
@@ -32,14 +33,16 @@ const SharingLink = React.createClass({
         if (!this.props.url) {
             return null;
         }
+        // add authentication to the url if possible
+        const url = SecurityUtils.addAuthenticationToUrl(this.props.url);
```

Comment:
This functionality have 2 purposes:
1. Get The link for the WMS for your personal GIS
2. Share with another user 
In the first case it is ok to skip authentication problems, but for the 2nd case is very bad to share the authentication key. 
So make it optional.